storage/replica_raftstorage.go

```diff
@@ -90,8 +88,39 @@ func (r *Replica) Entries(lo, hi, maxBytes uint64) ([]raftpb.Entry, error) {
 	// stopping once we have enough.
 	var ents []raftpb.Entry
 	size := uint64(0)
+
+	hitEnts, hitSize, hitIndex := r.raftEntryCache.getEntry(lo, hi, maxBytes)
+	// Did the correct number of results come back? If so, we're all good.
+	if len(hitEnts) == int(hi)-int(lo) {
+		if TestingEntriesCacheHook != nil {
+			r.Lock()
+			TestingEntriesCacheHook(hitEnts)
```

Comment:
Ah, right. I was assuming `entry_cache` would be populated on read, not just on append. But given the way entries are accessed, it makes sense that the cache would only be needed for newly-written entries. 

I'd still like to get rid of the hook, though. Let's move the cache-related functionality to `entriesFromCache()` and call it directly from the test (the test would call both `entriesFromCache()` and `Entries()`; the results from the cache would only be used to validate the number of entries returned.